### PR TITLE
Fix .gitpod.yml port name and description reflect delay

### PR DESCRIPTION
**Pull request by mustard-mh**
This PR fixes `Remote Explorer` ports name and description reflect delay when `.gitpod.yml` changed

It's caused by, `label` and `tooltip` is rely on the previous status of port. So we need to change `.gitpod.yml` twice to make prev works before.

## How to Test

Preview environment here https://hw-port-vs.preview.gitpod-dev.com/workspaces

1. Open a workspace like https://hw-port-vs.preview.gitpod-dev.com/#https://github.com/gitpod-io/template-sveltejs
2. Choose latest (insiders) code
3. Code commit hash should be `d66c73af75043dae5f76fd8655ecca195391ebe1` (in About of vscode)
4. Change `.gitpod.yml` to see if Remote Explorer reflect correctly

### CODE REVIEW

File: extensions/gitpod-web/src/extension.ts [hunk 1]
```diff
@@ -397,27 +397,29 @@ export class GitpodWorkspaceTreeDataProvider implements vscode.TreeDataProvider<
...
-				if (remotePort && remotePort !== portNumber) {
-					port.label += ':' + remotePort;
+				if (port.remotePort && port.remotePort !== localPort) {
+					port.label += ':' + port.remotePort;
 				}
 
-				const exposed = portStatus.getExposed();
 				const accessible = exposed || port.tunnel;
-				if (!portStatus.getServed()) {
+				const isPortTunnelPublic = port.tunnel?.privacy === 'public'; // TunnelPrivacyId
```
Comment (mustard-mh) [inline extensions/gitpod-web/src/extension.ts:421]: I don't know how to import `TunnelPrivacyId` here, code below not works
```
import { TunnelPrivacyId } from 'vs/platform/tunnel/common/tunnel';
```

Comment (jeanp413) [reply extensions/gitpod-web/src/extension.ts:421]: The values come from [here](https://github.com/gitpod-io/openvscode-server/blob/3dd5bb05411a5eb2a9ded6aca5d38d938fefaf7f/src/vs/gitpod/browser/workbench/workbench.ts#L801-L812) but seems we don't use it, only `public` [here](https://github.com/gitpod-io/openvscode-server/blob/3dd5bb05411a5eb2a9ded6aca5d38d938fefaf7f/src/vs/gitpod/browser/workbench/workbench.ts#L790-L794) so just continue using `public` :smile:

**Review by jeanp413 — APPROVED** [review]
LGTM, only need to fix this minor [comment](https://github.com/gitpod-io/openvscode-server/pull/360#discussion_r882953586)

**Comment by mustard-mh** [comment]
Updated, you can see diff [here](https://github.com/gitpod-io/openvscode-server/compare/d66c73af75043dae5f76fd8655ecca195391ebe1..aa0dc862996b8d58b4bc5bb8950c1c6e98cafd19)

**Pull request merged**